src/main/java/com/ericsson/ei/handlers/ExtractionHandler.java

```diff
@@ -78,23 +77,31 @@ public void runExtraction(RulesObject rulesObject, String mergeId, String event,
         try {
             JsonNode extractedContent = extractContent(rulesObject, event);
 
+            String mergedContent = null;
+            String aggregatedObjectId = null;
             if(aggregatedDbObject != null) {
                 LOGGER.debug("ExtractionHandler: Merging Aggregated Object:\n{}"
                         + "\nwith extracted content:\n{}"
                         + "\nfrom event:\n{}",
                         aggregatedDbObject.toString(), extractedContent.toString(), event);
-                String objectId = objectHandler.extractObjectId(aggregatedDbObject);
-                String mergedContent = mergeHandler.mergeObject(objectId, mergeId, rulesObject, event, extractedContent);
-                processRulesHandler.runProcessRules(event, rulesObject, mergedContent, objectId, mergeId);
+                aggregatedObjectId = objectHandler.extractObjectId(aggregatedDbObject);
+                mergedContent = mergeHandler.mergeObject(aggregatedObjectId, mergeId, rulesObject, event, extractedContent);
+                mergedContent = processRulesHandler.runProcessRules(event, rulesObject, mergedContent, aggregatedObjectId, mergeId);                
             } else {
+            	LOGGER.debug("***** Extraction starts for the Id: " + mergeId);
                 ObjectNode objectNode = (ObjectNode) extractedContent;
                 objectNode.put("TemplateName", rulesObject.getTemplateName());
-                mergeHandler.addNewObject(event, extractedContent, rulesObject);
+                mergedContent = mergeHandler.addNewObject(event, extractedContent, rulesObject);
+                aggregatedObjectId = mergeId;
```

Comment:
What is going on here? Why is the aggregatedObjectId updated to something else?

Reply:
This code we changed to make sure we call the runSubscription method only once for the aggregated object.
We had to make changes so that we can remove the dependency from PostInsetActions and changes are done in both if and else statements.

Reply:
I do understand the confusion. Maybe it would be good with a comment on why the aggregated id is the same as the merge id.